src/commands/apphosting-rollouts-create.ts

```diff
@@ -1,27 +1,116 @@
 import * as apphosting from "../gcp/apphosting";
-import { logger } from "../logger";
 import { Command } from "../command";
 import { Options } from "../options";
 import { needProjectId } from "../projectUtils";
+import { FirebaseError } from "../error";
+import * as ora from "ora";
+import { getRepoDetailsFromBackend, listAllBranches } from "../gcp/devConnect";
+import {
+  getGitHubBranch,
+  getGitHubCommit,
+  promptGitHubBranch,
+} from "../apphosting/githubConnections";
+import { orchestrateRollout } from "../apphosting";
+import { confirm } from "../prompt";
+import { logBullet } from "../utils";
+import { consoleOrigin } from "../api";
 
-export const command = new Command("apphosting:rollouts:create <backendId> <buildId>")
+export const command = new Command("apphosting:rollouts:create <backendId>")
   .description("create a rollout using a build for an App Hosting backend")
   .option("-l, --location <location>", "specify the region of the backend", "us-central1")
   .option("-i, --id <rolloutId>", "id of the rollout (defaults to autogenerating a random id)", "")
+  .option("-b, --git-branch <gitBranch>", "repository branch to deploy (defaults to 'main')", "")
+  .option("-c, --commit <commit>", "git commit to deploy (defaults to the latest commit)", "")
+  .option("-f, --force", "skip confirmation")
   .before(apphosting.ensureApiEnabled)
-  .action(async (backendId: string, buildId: string, options: Options) => {
+  .action(async (backendId: string, options: Options) => {
     const projectId = needProjectId(options);
     const location = options.location as string;
-    // TODO: Should we just reuse the buildId?
-    const rolloutId =
-      (options.buildId as string) ||
-      (await apphosting.getNextRolloutId(projectId, location, backendId));
-    const build = `projects/${projectId}/backends/${backendId}/builds/${buildId}`;
-    const op = await apphosting.createRollout(projectId, location, backendId, rolloutId, {
-      build,
+
+    let branch = options.gitBranch as string | undefined;
+    const commit = options.commit as string | undefined;
+    if (branch && commit) {
+      throw new FirebaseError(
+        "Cannot specify both a branch and commit to deploy. Please specify either --git-branch or --commit.",
+      );
+    }
+    const backend = await apphosting.getBackend(projectId, location, backendId);
+    if (!backend.codebase.repository) {
+      throw new FirebaseError(
+        `Failed to find connected Git repository for backend ${backendId}. Please check that your backend has been successfully linked to a Git repository.`,
```

Comment:
How do we make this error more actionable?

Can users easily associate new git repositories with existing backends? If not, consider prompting users to delete and recreate the backend

Reply:
OR maybe we say that the backend is misconfigured due to missing a repository, and prompt them to delete the backend (and call out to deleteBackend from here if they confirm)

Reply:
AFAIK we don't have a way for users to associate new git repositories with existing backends. My understanding is that backends _must_ be integrated with a git repository, and this is an invariant our control plane maintains (`backend.codebaes.repository` is a required field in the App Hosting API proto). 

So this is more of a "this should never happen, but if it does please let us know" scenario.